pkg/vault/vault.go

```diff
@@ -0,0 +1,176 @@
+package vault
+
+import (
+	"github.com/urfave/cli"
+
+	"fmt"
+	"io/ioutil"
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"net/url"
+	"net/http"
+	"time"
+
+//	"github.com/aws/aws-sdk-go/aws"
+//	"github.com/aws/aws-sdk-go/aws/credentials"
+//	"github.com/aws/aws-sdk-go/aws/credentials"
+//	"github.com/aws/aws-sdk-go/aws/session"
+	api "github.com/hashicorp/vault/api"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/reverbdotcom/rv/pkg/iam"
+)
+
+type RoleTokens map[string]Token
+
+type Token struct {
+	Token string `json:"token"`
+	Expiration time.Time `json:"expiration"`
+}
+
+func configFileName() string {
+	home := os.Getenv("HOME")
+	return home + "/.rv.json"
+	
+}
+
+func getConfig() RoleTokens {
+	var rt RoleTokens
+	raw, err := ioutil.ReadFile(configFileName())
+	if err != nil {
+		return rt
+	}
+
+	json.Unmarshal(raw, &rt)
+	return rt
+}
+
+func saveToken(role string, token Token) {
+	rt := getConfig()
+	if rt == nil {
+		rt = make(RoleTokens)
+	}
+	rt[role] = token
+	jsonOut, _ := json.Marshal(rt)
+	ioutil.WriteFile(configFileName(), jsonOut, 0400)
```

Comment:
We already have a cache for node lists -- you _could_ save the token in the same spot.